packages/context/src/__tests__/acceptance/context-view.acceptance.ts

```diff
@@ -0,0 +1,184 @@
+import {expect} from '@loopback/testlab';
+import {
+  Binding,
+  filterByTag,
+  Context,
+  ContextObserver,
+  ContextEventType,
+  ContextView,
+  Getter,
+  inject,
+} from '../..';
+
+let app: Context;
+let server: Context;
+
+describe('ContextView', () => {
+  let contextView: ContextView;
+  beforeEach(givenViewForControllers);
+
+  it('watches matching bindings', async () => {
+    // We have server: 1, app: 2
+    expect(await getControllers()).to.eql(['1', '2']);
+    server.unbind('controllers.1');
+    // Now we have app: 2
+    expect(await getControllers()).to.eql(['2']);
+    app.unbind('controllers.2');
+    // All controllers are gone from the context chain
+    expect(await getControllers()).to.eql([]);
+    // Add a new controller - server: 3
+    givenController(server, '3');
+    expect(await getControllers()).to.eql(['3']);
+  });
+
+  function givenViewForControllers() {
+    givenServerWithinAnApp();
+    contextView = server.createView(filterByTag('controller'));
+    givenController(server, '1');
+    givenController(app, '2');
+  }
+
+  function givenController(_ctx: Context, _name: string) {
+    class MyController {
+      name = _name;
+    }
+    _ctx
+      .bind(`controllers.${_name}`)
+      .toClass(MyController)
+      .tag('controller');
+  }
+
+  async function getControllers() {
+    // tslint:disable-next-line:no-any
+    return (await contextView.values()).map((v: any) => v.name);
+  }
+});
+
+describe('@inject.* - injects a live collection of matching bindings', async () => {
```

Comment:
Re-posting https://github.com/strongloop/loopback-next/pull/2122#discussion_r245930839

The name passed to `describe` is usually supposed to describe what is being tested (the subject under test), not the actual behavior/test scenarios.

Also `@inject.filter` was renamed to `@inject.view` IIUC.

Last but not least, test suites should not do any async computation, i.e. should be defined as a regular function (not an async function).

My proposal:

```suggestion
describe('@inject.view', () => {
```